src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,135 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
+@SuppressWarnings("unchecked")
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final Object[] EMPTY_ELEMENT_DATA = {};
+    private Object[] elementList;
+    private int size;
+
+    public ArrayList() {
+        this.elementList = EMPTY_ELEMENT_DATA;
```

Comment:
```suggestion
        elementList = EMPTY_ELEMENT_DATA;
```
https://stackoverflow.com/a/2411283